src/backend/opencl/err_opencl.hpp

```diff
@@ -14,26 +14,31 @@
 #include <platform.hpp>
 #include <types.hpp>
 
+#undef CATCHALL
+
+#define CATCHALL                                            \
+  catch(const cl::Error &err) {                             \
+    char opencl_err_msg[1024];                              \
+    snprintf(opencl_err_msg,                                \
+             sizeof(opencl_err_msg),                        \
+             "OpenCL Error (%d): %s when calling %s",       \
+             err.err(), getErrorMessage(err.err()).c_str(), \
+             err.what());                                   \
+    if (err.err() == CL_MEM_OBJECT_ALLOCATION_FAILURE) {    \
+      AF_ERROR(opencl_err_msg, AF_ERR_NO_MEM);              \
+    } else {                                                \
+      AF_ERROR(opencl_err_msg, AF_ERR_INTERNAL);            \
+    }                                                       \
+  }                                                         \
+  catch(...) {                                              \
+    return processException();                              \
+  }
+
```

Comment:
May be instead of doing `#undef CATCHALL` and defining the whole macro to include the original `CATCHALL` functionaliy, wouldn't it be better to do the following:

```
#define COMMON_CATCHAL CATCHALL
#undef CATCHALL

#define CATCHALL \
   catch (const cl::Error &err) {\
       //handle the error \
    } \
   COMMON_CATCHALL
```

This is to ensure any changes we make in `err_common.hpp` will be reflected immediately here.